driver-core/src/main/java/com/datastax/driver/core/ExecutionInfo.java

```diff
@@ -87,7 +88,10 @@ ExecutionInfo withIncomingPayload(Map<String, ByteBuffer> incomingPayload) {
      * {@link com.datastax.driver.core.policies.RetryPolicy} may retry the
      * query on the same host, so the same host might appear twice.</li>
```

Comment:
Just as a reminder, we might want to change the return type in version 4.0 to `Set<Host>`, because it doesn't really make sense to return a `List` since we introduced retries on the same host. Or better yet, some sort of `Map<SpeculativeExecution, Host>`. And even better yet, a custom structure `RequestTimeline` that would record all the relevant events that lead from the request submission to request completion (e.g. at instant T0 request was submitted, at instant T1 spec exec 1 was fired, at instant T2 normal exec timed out, etc...) – in other words, some sort of client-side query trace.